extensions/procfs/CpuStat.h

```diff
@@ -34,7 +34,12 @@ class CpuStatData {
  public:
   static std::optional<CpuStatData> parseCpuStatLine(std::istream& iss);
 
-  auto operator<=>(const CpuStatData& rhs) const = default;
+  bool operator<=(const CpuStatData& rhs) const;
+  bool operator>(const CpuStatData& rhs) const;
+  bool operator>=(const CpuStatData& rhs) const;
+  bool operator<(const CpuStatData& rhs) const;
+  bool operator==(const CpuStatData& rhs) const;
+  bool operator!=(const CpuStatData& rhs) const;
```

Comment:
Why this over the spaceship operator?

Reply:
Clang compilation failed with a warning on the spaceship operator in this particular case.
```
/home/ggyimesi/projects/nifi-minifi-cpp-fork/extensions/procfs/CpuStat.h:37:8: error: explicitly defaulted three-way comparison operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
  auto operator<=>(const CpuStatData& rhs) const = default;
       ^
/home/ggyimesi/projects/nifi-minifi-cpp-fork/extensions/procfs/CpuStat.h:57:23: note: defaulted 'operator<=>' is implicitly deleted because there is no viable comparison function for member 'user_'
  SystemClockDuration user_;
```

I'm not sure why it can't use it with the SystemClockDuration type.

Reply:
This is a clang (or rather libc++) bug. I would rather have one manual spaceship operator implementation that lexicographically compares each member instead of writing all six manually, without any kind of strict ordering.

Reply:
I agree, but if werror is turned on this fails the compilation. What would you suggest?

Reply:
Does it even make sense to say one CpuStatData is greater than another? If it's only for map storage, equality operator and hash would be enough, and turning the map into an unordered one.

Reply:
I suggest using memberwise lexicographical comparison in the simplest possible way. AND'ing together the comparison of all members is not consistent: there can be values that are both less and greater than each other, depending on the expression used.

example comparator: https://godbolt.org/z/nz553fbxG

Reply:
I had some success with a simpler workaround: just define operator<=> when using libc++.
See the branch on my fork: https://github.com/szaszm/nifi-minifi-cpp/tree/MINIFICPP-1832
Take the last commit if you like it.

Reply:
Thanks for the workaround! I added your commit to the PR 7c6ee4bbb1d0e7319821d4b5957fc41493b5f3b2